test/format/checksum.c

```diff
@@ -0,0 +1,136 @@
+#include "format.h"
+
+#define FNV_PRIME 0x00000100000001b3
+
+struct checksum_table_arg {
+    WT_SESSION *session;
+    uint64_t *hash;
+};
+
+/*
+ * fnv1a_init --
+ *     Initialize an incremental FNV-1A hash with a seed value.
+ */
+static void
+fnv1a_init(uint64_t *hash)
+{
+    *hash = 0xcbf29ce484222325;
+}
+
+/*
+ * fnv1a_add_bytes --
+ *     Update an incremental FNV-1A hash using an arbitrary run of bytes.
+ */
+static uint64_t
+fnv1a_add_bytes(uint64_t cur_hash, const uint8_t *data, size_t sz)
+{
+    for (size_t i = 0; i < sz; i++) {
+        cur_hash ^= data[i];
+        cur_hash *= FNV_PRIME;
+    }
+
+    return (cur_hash);
+}
+
+/*
+ * fnv1a_add_u32 --
+ *     Update an incremental FNV-1A hash using the four bytes of a u32.
+ */
+static uint64_t
+fnv1a_add_u32(uint64_t cur_hash, uint32_t data)
+{
+    for (int i = 0; i < 4; i++) {
+        cur_hash ^= data & 0xff;
+        cur_hash *= FNV_PRIME;
+
+        data >>= 8;
+    }
+
+    return (cur_hash);
+}
+
+/*
+ * checksum_key --
+ *     Update an incremental checksum with the key part of a key/value pair.
+ */
+static void
+checksum_key(uint64_t *hash, TABLE *table, WT_ITEM *key)
+{
+    uint32_t keyno = atou32("checksum-key", (char *)key->data + NTV(table, BTREE_PREFIX_LEN), '.');
+    *hash = fnv1a_add_u32(*hash, keyno);
+}
+
+/*
+ * checksum_value --
+ *     Update an incremental checksum with the value part of a key/value pair.
+ */
+static void
+checksum_value(uint64_t *hash, WT_ITEM *value)
+{
+    *hash = fnv1a_add_bytes(*hash, value->data, value->size);
+}
+
+/*
+ * checksum_table --
+ *     Update an incremental checksum with the contents of a table.
+ */
+static void
+checksum_table(TABLE *t, void *arg)
+{
+    struct checksum_table_arg *args = arg;
+
+    uint64_t *hash = args->hash;
+    WT_SESSION *session = args->session;
+    const char *uri = t->uri;
+
+    wt_wrap_begin_transaction(session, NULL);
```

Comment:
This will result in a long-running transaction, I am not sure of its consequence for read only ops.